src/main/java/com/liveramp/captain/daemon/CaptainJoblet.java

```diff
@@ -188,7 +189,13 @@ private void submitRequest(CaptainRequestConfig config) throws DaemonException {
       notifier.notify(subject, e, CaptainNotifier.NotificationLevel.ERROR);
       requestUpdater.cancel(id);
       return;
-    } catch (Throwable e) {
+    } catch(CaptainNamedException ce)
```

Comment:
I would prefer a more specific exception name than `CaptainNamedException`. Something similar to `CaptainPersistorException`, where the exception name describes the context. Something like `CaptainSubmissionException`?

Reply:
I agree. Thinking the same. Will update.